contracts/vesting/src/contract.rs

```diff
@@ -0,0 +1,636 @@
+            log!(
+                &env,
+                "Vesting: Initialize: At least one balance must be provided."
+            );
+            panic_with_error!(env, ContractError::MissingBalance);
+        }
+
+        let total_supply = create_vesting_accounts(&env, max_vesting_complexity, vesting_balances)?;
+        if let Some(mi) = minter_info {
+            let capacity = mi.capacity.value(env.ledger().timestamp()) as i128;
+            if total_supply > capacity {
+                log!(&env, "Vesting: Initialize: total supply over the capacity");
+                panic_with_error!(env, ContractError::SupplyOverTheCap);
+            }
+            save_minter(&env, mi);
+        }
+
+        let token_info = VestingTokenInfo {
+            name: vesting_token.name,
+            symbol: vesting_token.symbol,
+            decimals: vesting_token.decimals,
+            address: vesting_token.address,
+            total_supply,
+        };
+
+        save_whitelist(&env, &whitelisted_accounts);
+        save_token_info(&env, &token_info);
+        save_max_vesting_complexity(&env, &max_vesting_complexity);
+
+        env.events()
+            .publish(("Initialize", "Vesting contract with admin: "), admin);
+
+        Ok(())
+    }
+
+    fn transfer_token(
+        env: Env,
+        from: Address,
+        to: Address,
+        amount: i128,
+    ) -> Result<(), ContractError> {
+        from.require_auth();
+
+        if amount <= 0 {
+            log!(&env, "Vesting: Transfer token: Invalid transfer amount");
+            panic_with_error!(env, ContractError::InvalidTransferAmount);
+        }
+
+        let token_client = token_contract::Client::new(&env, &get_token_info(&env).address);
+
+        verify_vesting(&env, &from, amount, &token_client)?;
+        token_client.transfer(&from, &to, &amount);
+
+        env.events().publish(
+            (
+                "Transfer token",
+                "Transfering tokens between accounts: from: {}, to:{}, amount: {}",
+            ),
+            (from, to, amount),
+        );
+
+        Ok(())
+    }
+
+    fn transfer_vesting(
+        env: Env,
+        from: Address,
+        to: Address,
+        amount: i128,
+        curve: Curve,
+    ) -> Result<(), ContractError> {
+        from.require_auth();
+
+        let whitelist = get_whitelist(&env);
+        if !whitelist.contains(from.clone()) {
+            log!(
+                &env,
+                "Vesting: Transfer Vesting: Not authorized to transfer vesting"
+            );
+            panic_with_error!(env, ContractError::NotAuthorized);
+        }
+
+        if amount <= 0 {
+            log!(
+                &env,
+                "Vesting: Transfer Vesting: Transfer amount must be positive"
+            );
+            panic_with_error!(env, ContractError::InvalidTransferAmount);
+        }
+        curve.validate_monotonic_decreasing()?;
+        let (low, high) = curve.range();
+        if low != 0 {
```

Comment:
Please just add a comment that low means the starting value, meaning it can't be 0 since it needs to decrease from something.